src/main/drivers/barometer/barometer_icp201xx.c

```diff
@@ -0,0 +1,889 @@
+    lastFifoReadTime = millis();
+
+    // Setup function pointers
+    baro->combined_read = true;
+    baro->ut_delay = 0;
+    baro->up_delay = 25000;  // 25ms (25000µs) delay - read every 25ms to collect ~3 samples at 120Hz
+    baro->start_ut = icp201xxStartUT;
+    baro->get_ut = icp201xxGetUT;
+    baro->read_ut = icp201xxReadUT;
+    baro->start_up = icp201xxStartUP;
+    baro->get_up = icp201xxGetUP;
+    baro->read_up = icp201xxReadUP;
+    baro->calculate = icp201xxCalculate;
+
+    return true;
+}
+
+static bool icp201xxStartUT(baroDev_t *baro)
+{
+    UNUSED(baro);
+    return true; // Continuous mode, always running
+}
+
+static bool icp201xxReadUT(baroDev_t *baro)
+{
+    UNUSED(baro);
+    return true;
+}
+
+static bool icp201xxGetUT(baroDev_t *baro)
+{
+    UNUSED(baro);
+    return true;
+}
+
+static bool icp201xxStartUP(baroDev_t *baro)
+{
+    UNUSED(baro);
+    // Sensor is in continuous mode, always sampling
+    return true;
+}
+
+static bool icp201xxGetUP(baroDev_t *baro)
+{
+    // Check if bus is busy
+    if (busBusy(&baro->dev, NULL)) {
+        return false;
+    }
+
+    // State machine for non-blocking FIFO reads
+    switch (readState) {
+        case ICP201XX_STATE_IDLE:
+            // Do nothing, wait for readUP to start a read
+            break;
+
+        case ICP201XX_STATE_READ_FIFO_FILL_STARTED:
+            // FIFO fill read completed, copy data
+            fifoFillBuf[0] = spiRxBuf[0];
+
+            // Check if FIFO has data
+            if ((fifoFillBuf[0] & 0x40) == 0) {  // Not empty
+                uint8_t fifoCount = fifoFillBuf[0] & 0x1F;
+                if (fifoCount > 0 && fifoCount <= 16) {
+                    // Try reading ALL FIFO packets in one transaction (test auto-increment)
+                    fifoSampleCount = fifoCount;
+                    uint8_t bytesToRead = fifoCount * 6;
+
+                    if (icp201xxReadRegStart(&baro->dev, ICP201XX_REG_PRESS_DATA_0, bytesToRead)) {
+                        readState = ICP201XX_STATE_READ_FIFO_DATA_STARTED;
+                    } else {
+                        readState = ICP201XX_STATE_IDLE;
+                    }
+                } else {
+                    readState = ICP201XX_STATE_IDLE;
+                }
+            } else {
+                readState = ICP201XX_STATE_IDLE;
+            }
+            break;
+
+        case ICP201XX_STATE_READ_FIFO_DATA_STARTED:
+            // All FIFO data read completed in one transaction, copy to buffer
+            memcpy(fifoDataBuf, spiRxBuf, fifoSampleCount * 6);
+            readState = ICP201XX_STATE_DATA_READY;
+            break;
+
+        case ICP201XX_STATE_DATA_READY:
+            // Data already processed in calculate, ready for next read
+            break;
+    }
+
+    return true;
+}
+
+static bool icp201xxReadUP(baroDev_t *baro)
+{
+    // Rate limit reads
+    uint32_t now = millis();
+    if (now - lastFifoReadTime < ICP201XX_MIN_READ_INTERVAL_MS) {
```

Comment:
IT is slightly more readable to use (it limits range to 24days, but handles lastFifoReadTime in future better):

```suggestion
    if (cmp32_t(now, lastFifoReadTime) < ICP201XX_MIN_READ_INTERVAL_MS) {
```